src/routes/auth/login/+page.svelte

```diff
@@ -4,34 +4,113 @@
     Authentication-only pages should be redirected here if no user session exists.
     @see https://kit.svelte.dev/docs/form-actions
     */
-  import { redirect } from '@sveltejs/kit';
+  import { redirect } from "@sveltejs/kit";
 
-  /** @type {import('./$types').PageData} */
+  /** @type {import("./$types").PageData} */
   export let data;
 
-  /** @type {import('./$types').ActionData} */
+  /** @type {import("./$types").ActionData} */
   export let form;
 
+
   if (data.user) {
-    // If the user is already logged in, redirect them to the home page
-    throw redirect(307, '/');
+    console.log("Logged in already");
+    throw redirect(302, "/posts");
+    window.location.href = "/posts";
   }
+
+
+  // Handle Form submission by user
+  async function handleSubmit() {
+    console.log("handleSubmit called");
+    const username = usernameInput.value;
+    const password = passwordInput.value;
+    console.log("username:", username, "password:", password);
+
+    // Send POST request to server with provided credentials
+    const response = await fetch("/auth/login", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded"
+      },
+      body: new URLSearchParams({
+        username: username,
+        password: password
+      })
+    });
+
+    console.log("response:", response);
+    console.log("response status:", response.status);
+
+
+    // Server response OK = Handle the login successfully
+    if (response.ok) {
+      console.log("User has logged in");
+      const responseData = await response.json();
+      const dataString = responseData.data;
+      data = JSON.parse(dataString);
+
+      console.log("data:", data);
+
+      const returnMessage = data[2];
+      const jsonRM = JSON.parse(returnMessage);
+      console.log(jsonRM.message);
+      console.log();
+
+      successMessage = jsonRM.message;
+
+      // Redirect user to the homepage after a successful login
+      // TODO: Add to user variable the user's class/info
+      window.location.href = "/";
+
+    } else {
+      // If the server response is not ok, display an error message.
+      const error = await response.json();
+      console.log("error:", error);
+      console.log("error.message:", error.body);
+      errorMessage = error.body;
+    }
+
+  }
```

Comment:
This whole function is unnecessary. Form submissions are handled by Svelte, you do not need to manually POST data - I'd suggest reading over the whole of https://kit.svelte.dev/docs/form-actions

Reply:
Using svelte's implementations kept giving me issues with parsing the data back to the server in an appropriate format - will look into it further